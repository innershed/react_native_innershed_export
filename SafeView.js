import React from 'react';
import {View, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';
import Constants from 'expo-constants';

class SafeView extends React.Component {
    static propTypes = {
        children: PropTypes.node,
        style: PropTypes.any
    };

    constructor(props) {
        super(props);
    }

    render() {
        const {children, style, ...rest} = this.props;

        return (
            <View style={styles.container}>
                <View style={styles.bar}/>
                <View {...rest} style={[style, styles.content]}>
                    {children}
                </View>
            </View>
        );
    }
}

export default SafeView;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'stretch'
    },
    bar: {
        height: Constants.statusBarHeight,
        width: '100%'
    },
    content: {
        width: '100%',
        position: 'absolute',
        top: Constants.statusBarHeight,
        bottom: 0
    }
});
