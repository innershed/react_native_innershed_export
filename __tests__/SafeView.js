import React from 'react';
import { Text } from 'react-native';

import renderer from 'react-test-renderer';

import SafeView from '../SafeView';

it('renders', () => {
    expect(renderer.create(
        <SafeView>
            <Text>Foobar</Text>
        </SafeView>
    )).toMatchSnapshot();
});
