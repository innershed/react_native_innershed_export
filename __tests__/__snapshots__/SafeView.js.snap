// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 20,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Array [
        undefined,
        Object {
          "bottom": 0,
          "position": "absolute",
          "top": 20,
          "width": "100%",
        },
      ]
    }
  >
    <Text>
      Foobar
    </Text>
  </View>
</View>
`;
